src/main/java/com/mixpanel/android/mpmetrics/PersistentIdentity.java

```diff
@@ -574,6 +624,35 @@ private void readIdentities() {
         mIdentitiesLoaded = true;
     }
 
+    private void readOptOutFlag() {
+        SharedPreferences prefs = null;
+        try {
+            prefs = mMixpanelPreferences.get();
+        } catch (final ExecutionException e) {
+            MPLog.e(LOGTAG, "Cannot read opt out flag from sharedPreferences.", e.getCause());
+        } catch (final InterruptedException e) {
+            MPLog.e(LOGTAG, "Cannot read opt out flag from sharedPreferences.", e);
+        }
+
+        if (null == prefs) {
```

Comment:
well.. i might be too picky but this is not how to do this kind of check at other places.
eg. `if (mIsUserOptOut == null)`